File: HandOnML/P02_Regression/06_RandomForestRegression/random_forest_regression.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[,2:3]

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$DependentVariable, SplitRatio = 0.8)
# training_set = subset(dataset, split == TRUE)
# test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)

# Fitting Regression to the dataset
#install.packages('randomForest')
library(randomForest)
set.seed(1234)
regressor = randomForest(x = dataset[1], # dataframe
                         y = dataset$Salary, # Vector
                         ntree = 500) # Number of trees to grow

# Predict a new result with Regression
predict(regressor, data.frame(Level = 6.5))

# Visualising the Random Forest Regression modal with smooth curve
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
x_grid_frame = data.frame(Level = x_grid,
            Level2 = x_grid^2,
            Level3 = x_grid^3,
            Level4 = x_grid^4)
ggplot() +
    geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') +
    geom_line(aes(x = x_grid, y = predict(regressor, newdata = x_grid_frame)), color = 'blue') +
    ggtitle('Truth or Bluff (Random Forest Regression)') +
    xlab('Position Level') +
    ylab('Salary')

File: HandOnML/P02_Regression/06_RandomForestRegression/random_forest_regression.py
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
# X must be matrix not a vector. So we need to specify the range of collumn [1:2]
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Splitting the dataset into the Training set and Test set
"""from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)"""


# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting the regression to the dataset
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators = 10, random_state = 0);
regressor.fit(X,y)

# Visualising the Random forest Regression Result for higher resolution and a smoother curve
X_grid = np.arange(min(X),max(X),0.1)
X_grid = X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color='red')
plt.plot(X_grid,regressor.predict(X_grid),color='blue')
plt.title('Truth or Bluff (Random forest regression)')
plt.xlabel('Position')
plt.ylabel('Salary')
plt.show()
